advanced/matrix/test_matrix.py

```diff
@@ -0,0 +1,166 @@
+"""Unit testing for the Matrix class."""
+import unittest
+
+from matrix import Matrix
+
+
+class TestMatrix(unittest.TestCase):
+    """Test the initialization and methods of the matrix"""
+
+    def setUp(self):
+        """Initialize matrices a and b before tests."""
+        self.mat_a = Matrix(((1, 2), (3, 4)))
+        self.mat_b = Matrix(((3, 5), (6, 8)))
+
+    def test_tuples_property(self):
+        """Test tuples property of matrix."""
+        self.assertEqual(self.mat_a.matrix_rows, ((1, 2), (3, 4)))
+
+    def test_type_validation(self):
+        """Test that only a tuple of tuples is allowed in Matrix."""
+        with self.assertRaises(TypeError):
+            Matrix(((1, 2, 3), [4, 5, 6], 5))
+
+    def test_value_validation(self):
```

Comment:
maybe test_invalid_constructor or something like this.
its not clear from the name what you are testing